### DISCUSSION TITLE
Transforming SINGLE VIEW MRI images

### DISCUSSION THREAD

**Discussion by cepa995** [TorchIO-project/torchio#689] [Q&A] [answered]
Hey there!

Recently I've tried extracting AXIAL view from RSNA MICCAI Brain Tumor Kaggle competition, and storing it as a sequence of DICOM files by using DCMTK img2dcm binary. When I try to feed the data to my network using following set of preprocessing transformations:
```python
preprocess = tio.Compose([
            tio.ToCanonical(),
            tio.HistogramStandardization(landmarks_dict),
            tio.RescaleIntensity((-1, 1)),
            tio.Resample((1.0, 1.0, 1.0)),
            tio.Resize((128, 128, 64)),
        ])
```

I am getting this WARNING message which I ma not sure why it occurs:

`C:\Users\StefanCepa995\anaconda3\envs\kaggle\lib\site-packages\torchio\transforms\preprocessing\spatial\resize.py:55: UserWarning: Output shape (128, 128, 1) != target shape (128, 128, 64). Fixing with CropOrPad`

Why is output shape 128, 128, 1? How does keeping only 1 image VIEW affect this?

**Comment by fepegar** [comment, marked as answer]
Hi, @cepa995. Can you share one image to reproduce this?

`Resize` is typically not a good choice in medical imaging. Can you try with `CropOrPad` instead? `Resize`will deform your images.

**Reply by cepa995** [reply]
Well, that would explain a lot. Give me a minute.

**Reply by cepa995** [reply]
Ah, never mind. I think you were right @fepegar about Resize. It really ruins everything. I think you solved my issue with CropOrPad. Thanks!

**Reply by fepegar** [reply]
Thanks for your feedback. I need to add a warning in the docs for `Resize`!

**Reply by cepa995** [reply]
Yeah, that would be great! Thanks!

**Reply by cepa995** [reply]
@fepegar  out of curiosity, is there any built-in functionality in torch IO to train NN on single image view like AXIAL? I did code something up myself, but I am wondering if there is already an in-built mechanism to achieve this.

**Reply by fepegar** [reply]
You could use patch-based training. Take a look at the [Sample slices from volumes](https://torchio.readthedocs.io/auto_examples/plot_3d_to_2d.html#sample-slices-from-volumes) example in the [gallery](https://torchio.readthedocs.io/auto_examples/index.html).
